crates/nu-parser/src/hir/path.rs

```diff
@@ -39,3 +39,105 @@ impl Path {
         (self.head, self.tail)
     }
 }
+
+// WIP: find a better place for these functions.
```

Comment:
Open to suggestions about where to place the various functions

Reply:
Looking at the changes, this information is used in 2 distinct places:

1. crates/nu-parser/src/hir/syntax_shape/expression.rs
2. src/commands/classified/external.rs

In this MR I'm adding the main implementation in crates/nu-parser/src/hir/path.rs and exposing it, this results in uses like `crate::hir::path::expand_tilde;` vs `use nu_parser::expand_tilde;`

None of of them seems optimal, do you have some suggestion where to put this code?